tracer/src/Datadog.Trace/Agent/Api.cs

```diff
@@ -299,8 +299,14 @@ private async Task<bool> SendTracesAsyncImpl(IApiRequest request, bool finalTry,
 
                 TelemetryFactory.Metrics.RecordCountTraceApiResponses(response.GetTelemetryStatusCodeMetricTag());
 
-                // Attempt a retry if the status code is not SUCCESS
-                if (response.StatusCode < 200 || response.StatusCode >= 300)
+                // A change on the agent changed the default response when payloads are dropped from being 200 to 429
+                // This change would cause us to change behavior and start retrying those requests which were
+                // being rate limited due to performance issues on the agent.
+                // To maintain the same behavior for the Tracer we'll treat 429 as a "success" for the time being
+                // to allow us to determine a better course of action.
+                // https://github.com/DataDog/datadog-agent/pull/17917
+                // Attempt a retry if the status code is not SUCCESS and NOT 429
+                if ((response.StatusCode < 200 || response.StatusCode >= 300) && response.StatusCode != 429)
```

Comment:
I think we should update this so that we still log that the requests failed, but that we didn't retry. Currently we won't have any visibility that this happened.

It is probably sufficient to keep this line here, and then conditionally change the  log message on line 360, e.g.

```csharp
if(response.StatusCode == 429)
{
     var retryAfter = response.GetHeader("Retry-After");
    _log.Debug<int, string>("Failed to submit {Count} traces. Agent responded with 429 Too Many Requests, retry after {RetryAfter}", numberOfTraces, retryAfter ?? "unspecified");
}
else
{
    _log.Debug<int>("Successfully sent {Count} traces to the Datadog Agent.", numberOfTraces);
}
```

I'm not sure if they include the `Retry-After` header, but I expect so. Also, if there's any response content, it _may_ be worth including it.

Reply:
Thanks I'll add those!

Reply:
Added the `retryAfter` and debug log